src/org/pentaho/reporting/platform/plugin/cache/FileSystemCacheBackend.java

```diff
@@ -51,50 +51,82 @@ public void setCachePath( final String cachePath ) {
 
   @Override
   public boolean write( final List<String> key, final Serializable value ) {
-    final File file = new File( cachePath + StringUtils.join( cleanKey( key ), File.separator ) );
-
-    final ObjectOutputStream oos;
-    final FileOutputStream fout;
+    final String filePath = cachePath + StringUtils.join( cleanKey( key ), File.separator );
+    final File file = new File( filePath );
+    lock( filePath );
     try {
+      //create file structure
       file.getParentFile().mkdirs();
       if ( !file.exists() ) {
         file.createNewFile();
       }
-
-      fout = new FileOutputStream( file );
-      oos = new ObjectOutputStream( fout );
-      oos.writeObject( value );
-      oos.close();
-      fout.close();
+      //closable resources
```

Comment:
Ahh, yeah. The curly brace and round braces look awfully the same on githubs font.